src/colorizer/FrameCache.ts

```diff
@@ -1,48 +1,54 @@
-import { Texture } from "three";
-
-type CacheEntry = {
-  frame: Texture;
-  index: number;
-  prev: CacheEntryNullable;
-  next: CacheEntryNullable;
+type DisposableValue = {
+  dispose?: () => void;
 };
 
-type CacheEntryNullable = CacheEntry | null;
+type CacheEntry<E extends DisposableValue> = {
+  key: string;
+  value: E;
+  size: number;
+  prev: CacheEntry<E> | null;
+  next: CacheEntry<E> | null;
+};
 
-/** LRU cache for textures. Calls `dispose` on eviction to keep GPU memory under control. */
-export default class FrameCache {
-  private data: CacheEntryNullable[];
-  private first: CacheEntryNullable;
-  private last: CacheEntryNullable;
-  private numItems: number;
+/**
+ * Generic LRU cache for data, intended for Texture or other GPU resources.
+ * Calls `dispose` on eviction to keep GPU memory under control.
```

Comment:
curious from reading this comment: what do you do if the gpu needs a resource but it's been evicted?  

I guess you're going on the assumption that the most recently used set of assets required by the gpu for a given rendered frame will always fit in cache.